Vinyl/URLSessionDataTask.swift

```diff
@@ -8,23 +8,105 @@
 
 import Foundation
 
-public final class URLSessionDataTask: NSURLSessionDataTask, URLSessionTaskType {
-    
-    private let completion: Void -> Void
-    
-    init(completion: Void -> Void) {
-        self.completion = completion
-    }
-    
-   public override func resume() {
-        completion()
-    }
-    
-   public override func suspend() {
+private func generate() -> AnyGenerator<Int> {
+    var current = 0
+    return AnyGenerator<Int>(body: {
+        current = current + 1
+        return current
+    })
+}
+
+private let sharedSequence: AnyGenerator<Int> = generate()
+
+typealias URLSessionDataTaskCallback = (NSData?, NSURLResponse?, NSError?) -> Void
+
+public final class URLSessionDataTask: NSURLSessionDataTask {
+
+    private let session: Turntable
+    private let request: NSURLRequest
+    private let delegate: NSURLSessionDataDelegate?
+    private let callback: URLSessionDataTaskCallback?
+
+    init(session: Turntable, request: NSURLRequest, callback: URLSessionDataTaskCallback? = nil) {
+        self.session = session
```

Comment:
You pass the session as a constructor, but what you are really interested in, are its properties, this is called feature envy and I am not really keen on doing this. It's preferable to pass what you need to the `URLSessionDataTask`. I would also prefer to keep the internals of the Turntable private (you changed them to internal).